core/src/test/java/io/questdb/griffin/KeyedAggregationTest.java

```diff
@@ -1095,6 +1097,32 @@ public void testFirstLastAggregationsNotSupported() {
         }
     }
 
+    @Test
+    public void testStrFunctionKey() throws Exception {
+        // An important aspect of this test is that both replace() and count_distinct()
+        // functions use the same column as the input.
+        assertQuery("replace\tcount\tcount_distinct\n" +
+                        "foobaz\t63\t2\n" +
+                        "bazbaz\t37\t1\n",
+                "select replace(s, 'bar', 'baz'), count(), count_distinct(s) from tab",
+                "create table tab as (select timestamp_sequence(0, 100000) ts, rnd_str('foobar','foobaz','barbaz') s from long_sequence(100))",
+                null, true, true, true
+        );
+    }
+
+    @Test
+    public void testStrFunctionKeyReverseOrder() throws Exception {
```

Comment:
this test does not fail on master, is this just an additional test or was it aimed to reproduce the problem this PR addresses?

Reply:
It's just a symmetrical test aimed to catch possible future regressions.